src/toTree.js

```diff
@@ -314,15 +320,19 @@ export const toTree = (simple, version) => {
             throw new InvalidInputError('Unrecognized simple condition: ' + condition.Comparator.value);
         }
 
-        const { match, values } = buildMatchAndValues(comparator, condition);
-
+        const { match, values: matchValues } = buildMatchAndValues(comparator, condition);
+        const values = matchValues.map(escapeVariables);
         const conditionMap = {
             sender: () => buildAddressTest(['From'], values, match),
             recipient: () => buildAddressTest(['To', 'Cc', 'Bcc'], values, match),
             subject: () => buildSimpleHeaderTest(['Subject'], values, match),
             attachments: () => TEST_NODES.attachment[0]
         };
 
+        if (values.some(value => typeof value !== 'string')) {
```

Comment:
no need for a if.
`dollarNeeded = values.some(value => typeof value !== 'string')`